libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx

```diff
@@ -0,0 +1,383 @@
+import ErrorRenderer from './ErrorRenderer' // eslint-disable-line
+const StaticAnalysisRunner = require('@remix-project/remix-analyzer').CodeAnalysis
+const utils = remixLib.util
+
+/* eslint-disable-next-line */
+export interface RemixUiStaticAnalyserProps {
+  renderStaticAnalysis: any
+  staticanalysis: any
+  analysisRunner: any,
+  lastCompilationResult: any,
+  lastCompilationSource: any,
+  registry: any,
+  event: any,
+  analysisModule: any
+  _deps: any,
+  emit: any
+}
+
+export const RemixUiStaticAnalyser = (props: RemixUiStaticAnalyserProps) => {
+  const [runner] = useState(new StaticAnalysisRunner())
+
+  const preProcessModules = (arr: any) => {
+    return arr.map((Item, i) => {
+      const itemObj = new Item()
+      itemObj._index = i
+      itemObj.categoryDisplayName = itemObj.category.displayName
+      itemObj.categoryId = itemObj.category.id
+      return itemObj
+    })
+  }
+
+  const groupedModules = utils.groupBy(
+    preProcessModules(runner.modules()),
+    'categoryId'
+  )
+
+  const getIndex = (modules, array) => {
+    Object.values(modules).map((value: {_index}) => {
+      if (Array.isArray(value)) {
+        value.forEach((x) => {
+          array.push(x._index.toString())
+        })
+      } else {
+        array.push(value._index.toString())
+      }
+    })
+  }
+
+  const groupedModuleIndex = (modules) => {
+    const indexOfCategory = []
+    if (!_.isEmpty(modules)) {
+      getIndex(modules, indexOfCategory)
+    }
+    return indexOfCategory
+  }
+
+  const [categoryIndex, setCategoryIndex] = useState(groupedModuleIndex(groupedModules))
+
+  const warningContainer = React.useRef(null)
+  const [runButtonState, setRunButtonState] = useState(true)
+  const [autoRun, setAutoRun] = useState(false)
+  const [result, setResult] = useState({
+    lastCompilationResult: null,
+    lastCompilationSource: null,
+    currentFile: 'No file compiled'
+  })
+  const [, setModuleNameResult] = useState(null)
+  const [, setWarning] = useState({
+    msg: '',
+    options: {},
+    hasWarning: false,
+    warningErrors: []
+  })
+  const [warningState, setWarningState] = useState([])
+
+  useEffect(() => {
+    
+    if (autoRun) {
+      const setCompilationResult = async (data, source, file) => {
+        await setResult({ lastCompilationResult: data, lastCompilationSource: source, currentFile: file })
+      }
+      
+      if (props.analysisModule) {
+       
+        props.analysisModule.on(
+          'solidity',
+          'compilationFinished',
+          (file, source, languageVersion, data) => {
+            if (languageVersion.indexOf('soljson') !== 0) return
+            setCompilationResult(data, source, file)
+            if(categoryIndex.length > 0){
+              run(data, source, file)
+            }
+
+          }
+        )
+      }
+    } else {
+      setAutoRun(true)
+    }
+
+    return () => { }
+  }, [autoRun, categoryIndex])
```

Comment:
Do you want to activate autoRun if categoryIndex changes?

Reply:
no. have removed it